linux/NvidiaJetson.c

```diff
@@ -0,0 +1,321 @@
+
+The code tries to find out the correct sensors during the application startup. As an example, the sensors
+location for NVIDIA Jetson Orin:
+- CPU temperature: /sys/devices/virtual/thermal/thermal_zone0/type
+- GPU temperature: /sys/devices/virtual/thermal/thermal_zone1/type
+- GPU frequency: /sys/class/devfreq/17000000.gpu/cur_freq
+- GPU curr load: /sys/class/devfreq/17000000.gpu/device/load
+
+Measure:
+- The GPU frequency is provided in Hz, shown in MHz.
+- The CPU/GPU temperatures are provided in Celsius multipled by 1000 (milli Celsius), shown in Cesius
+- The Farenheit support is not provided
+
+If htop starts with root privileges (effective user id is 0), the experimental code activates.
+It reads the fixed sysfs file /sys/kernel/debug/nvmap/iovmm/clients with the following content, e.g.:
+```
+CLIENT                        PROCESS      PID        SIZE
+user                         gpu_burn     7979   23525644K
+user                      gnome_shell     8119       5800K
+user                             Xorg     2651      17876K
+total                                            23549320K
+```
+Unfortunately, the /sys/kernel/debug/... files are allowed to read only for the root user, that's why the restriction applies. 
+
+The patch out of this file adds a separate field 'GPU_MEM', which reads data from LinuxProcess::gpu_mem field.
+The field stores memory allocated for GPU in kilobytes. It is populated by the function
+NvidiaJetson_LoadGpuProcessTable (the implementation is located here), which is called at the end of the function
+Machine_scanTables.
+
+Additionally, the new Action is added: actionToggleGpuFilter, which is activated by 'g' hot key (the help is updated
+appropriately). The GpuFilter shows only the processes which currently utilize GPU (i.e. highly extended
+nvmap/iovmm/clients table). It is achieved by the filtering machinery associated with ProcessTable::pidMatchList.
+The code below constructs GPU_PID_MATCH_LIST hash table, then actionToggleGpuFilter either stores it to the
+ProcessTable::pidMatchList or restores old value of ProcessTable::pidMatchList.
+
+The separate LinuxProcess's PROCESS_FLAG_LINUX_GPU_JETSON (or something ...) flag isn't added for GPU_MEM, because
+currently the functionality of population LinuxProcess::gpu_mem is shared with the GPU consumers filter construction.
+So, even if GPU_MEM field is not activated, the filter showing GPU consumers should work. This kind of architecture is
+chosen intentially since it saves memory for the hash table GPU_PID_MATCH_LIST (which is now actually a set), and therefore
+increases performance. All other approaches convert GPU_PID_MATCH_LIST to a true key/value storage (key = pid,
+value = gpu memory allocated) with further merge code.
+
+*/
+
+
+/* global paths per each sensor */
+char CPU_TEMP_SENSOR_PATH[64];
+char GPU_TEMP_SENSOR_PATH[64];
+char GPU_FREQ_SENSOR_PATH[64];
+char GPU_LOAD_SENSOR_PATH[64];
+
+
```

Comment:
Highlevel blocks only have one blank between them. Cf. style guide. Only exception is directly after the headers, where it's two blanks.